fdbclient/NativeAPI.actor.cpp

```diff
@@ -1936,6 +1938,36 @@ Future<Optional<Value>> Transaction::get( const Key& key, bool snapshot ) {
 	if( !snapshot )
 		tr.transaction.read_conflict_ranges.push_back(tr.arena, singleKeyRange(key, tr.arena));
 
+	if(key == metadataVersionKey) {
+		if(!ver.isReady() || metadataVersion.isSet()) {
+			return metadataVersion.getFuture();
+		} else {
+			if(ver.isError()) return ver.getError();
+			if(ver.get() == cx->metadataVersionCache[cx->mvcInsertLocation].first) {
+				return cx->metadataVersionCache[cx->mvcInsertLocation].second;
+			}
+
+			Version v = ver.get();
+			int hi = cx->mvcInsertLocation;
+			int lo = (cx->mvcInsertLocation+1)%cx->metadataVersionCache.size();
+
+			while(true) {
+				int cu = hi > lo ? (hi + lo)/2 : ((hi + cx->metadataVersionCache.size() + lo)/2)%cx->metadataVersionCache.size();
+				if(v == cx->metadataVersionCache[cu].first) {
+					return cx->metadataVersionCache[cu].second;
+				}
+				if(cu == lo) {
+					break;
```

Comment:
This sort of works without the `READ_SYSTEM_KEYS` option, but if it hits this line and has to read through, I think it will fail without the option. It may be good to have the behavior be consistent to avoid surprises.

Reply:
NativeAPI transactions actually do not check for ACCESS_SYSTEM_KEYS, so it should be fine

Reply:
Oh, good point